### PR TITLE
Settings search refactor

### PR DESCRIPTION
**Pull request by hmalik88**
Refactored settings search code.

The purpose of this refactor is to introduce a cleaner, more efficient implementation of settings search. As it stands, the route objects getter function is creating a new array of route objects every time it is called. The idea is to extract something that should be a constant into its own file for easy updating in the future and to introduce some caching when it comes to calculating a filtered array of route objects. This update also facilitates in tackling issue #13672. Unnecessary util functions were removed and logic for feature flagged settings was simplified to a `featureFlag` property on the routeObject itself. This refactor is also making it easier to implement snaps view search.

**Note:** This also solves the following bugs: 

1. Ampersands weren't allowed in search input (when there's a setting that uses an ampersand)
2. Duplicate entries show in search results for certain settings when typing in an exact match into search input.

### CODE REVIEW

File: ui/helpers/utils/settings-search.js [hunk 1]
```diff
@@ -1,420 +1,54 @@
...
+    }
+    return true;
+  });
+  return settingsRoutes;
 }
 
-function getFilteredSettingsRoutes(t, tabName) {
-  return getSettingsRoutes(t).filter((s) => s.tab === tabName);
+function getFilteredSettingsRoutes(tabName) {
+  return getSettingsRoutes().filter((s) => s.tab === tabName);
```
Comment (rekmarks) [inline ui/helpers/utils/settings-search.js:20]: Single-character variable names are bad and we should do something different, like in my suggestion above.

File: ui/helpers/constants/settings.js [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+  ABOUT_US_ROUTE,
+  NETWORKS_ROUTE,
+  CONTACT_LIST_ROUTE,
+  EXPERIMENTAL_ROUTE,
+  ///: BEGIN:ONLY_INCLUDE_IN(flask)
+  SNAPS_LIST_ROUTE,
+  ///: END:ONLY_INCLUDE_IN
+} from './routes';
+
+export const SETTINGS_ROUTES = [
```
Comment (rekmarks) [inline ui/helpers/constants/settings.js:15]: These aren't just routes. Maybe something like SETTINGS_CONSTANTS would be better?

**Review by rekmarks — DISMISSED** [review]
Looks good! Did just a quick review, we'll dig in further tomorrow.

File: ui/helpers/constants/settings.js (hunk 2 above)
Comment (hmalik88) [reply ui/helpers/constants/settings.js:15]: sure

File: ui/pages/settings/settings-search-list/settings-search-list.js [hunk 3]
```diff
@@ -29,32 +29,32 @@ export default function SettingsSearchList({ results, onClickSetting }) {
...
                     'settings-page__header__search__list__item__tab',
                     {
                       'settings-page__header__search__list__item__tab-multiple-lines':
-                        tab === t('securityAndPrivacy'),
+                        tabKey === 'securityAndPrivacy',
                     },
                   )}
                 >
-                  {tab}
+                  {t(tabKey)}
```
Comment (brad-decker) [inline ui/pages/settings/settings-search-list/settings-search-list.js:41]: This is problematic, and the reason that the original implementation had passed the t method around. Setting variables to message keys and passing those around become a nightmare for figuring out how  messages are being used, and they result in our strict evaluation of where message keys are being used to fail.

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: ui/pages/settings/settings-search-list/settings-search-list.js (hunk 3 above)
Comment (hmalik88) [reply ui/pages/settings/settings-search-list/settings-search-list.js]: Cool, made some changes

**Review by brad-decker — DISMISSED** [review]
Very nice

**Comment by danjm** [comment]
Can you put some comments in the description of the motivation for this refactor and a summary of the changes (and where appropriate, the benefit/impact of the changes)?

**Comment by hmalik88** [comment]
@danjm updated the description.

**Review by rekmarks — APPROVED** [review]
LGTM!

**Review by brad-decker — APPROVED** [review]

**Pull request merged**